src/Common/tests/System/Net/Http/GenericLoopbackServer.cs

```diff
@@ -85,5 +89,35 @@ public int GetHeaderValueCount(string headerName)
         {
             return Headers.Where(h => h.Name.Equals(headerName, StringComparison.OrdinalIgnoreCase)).Count();
         }
+
+        public Span<byte> Body => _body.AsSpan().Slice(0, _bodyLength);
+
+        internal void AddBodyData(ReadOnlySpan<Byte> data)
```

Comment:
I don't think we should add this logic here. HttpRequestData is intentionally not very smart. Just expose something like `public Memory<byte> Body>` similar to method and path above. This logic for constructing the body should live in Http2LoopbacKServer.